ellama.el

```diff
@@ -2147,6 +2167,41 @@ otherwise prompt user for URL to summarize."
       :false-object nil)
      :same)))
 
+(defun ellama-generate-completion-list ()
+  "Generate completion list from context at point."
+  (mapcar (lambda (s)
+	    (car (reverse (string-split s))))
+	  (plist-get
+	   (json-parse-string
+	    (llm-chat
+	     (or ellama-completion-provider
+		 ellama-provider
+		 (ellama-get-first-ollama-chat-model))
+	     (llm-make-chat-prompt
+	      (if (region-active-p)
+		  (buffer-substring-no-properties (region-beginning) (region-end))
+		(buffer-substring-no-properties (point-min) (point)))
+	      :context ellama-completion-list-prompt-template
+	      :response-format '(:type object :properties
+				       (:completions (:type array :items (:type string)))
+				       :required (completions))))
+
+	    :object-type 'plist
+	    :false-object nil
+	    :array-type 'list)
+	   :completions)))
+
+(defun ellama-complete-at-point (&rest _)
+  "Complete at point using ellama."
+  (interactive)
+  (while-no-input
```

Comment:
`while-no-input` returns t when interrupted, so this seems not ideal. UIs would try to ignore errors, of course, but you might still see them in *Messages*.

For company specifically, you can try the `:company-use-while-no-input t` property instead. Corfu just aborts all completion tables on input, though, so it would be compatible with it anyway.